source/jobs/JobsFeature.cpp

```diff
@@ -43,9 +43,9 @@ void JobsFeature::ackSubscribeToNextJobChanged(int ioError)
     if (ioError)
     {
         // TODO We need to implement a strategy for what do when our subscription fails
-        string errorMessage = "Failed to subscribe to nextJobChanged";
+        string errorMessage = "Encountered an ioError while attempting to subscribe to NextJobChanged";
         LOG_ERROR(TAG, errorMessage.c_str());
```

Comment:
Also consider printing the error coming from IoT core. It will help user to debug the issue more easily. You can use "ErrorDebugString" method present in CRT SDK for this.

Reply:
I added logging of the specific io error in my latest revision, but I don't agree that we can use "ErrorDebugString" provided by the CRT SDK. I believe that method is reserved for printing readable versions of CRT SDK errors, but I'm pretty sure it won't make sense to have it try to interpret an error passed down from the Jobs service and/or the message broker.

Reply:
I think we can use it. I have seen it is being used in various [sample](https://github.com/aws/aws-iot-device-sdk-cpp-v2/blob/main/samples/shadow/shadow_sync/main.cpp#L90) codes.